src/main/java/alfred/parser/Parser.java

```diff
@@ -0,0 +1,58 @@
+package alfred.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import alfred.command.AddCommand;
+import alfred.command.Command;
+import alfred.command.DeleteCommand;
+import alfred.command.ExitCommand;
+import alfred.command.FindCommand;
+import alfred.command.ListCommand;
+import alfred.command.ListDateCommand;
+import alfred.command.MarkCommand;
+import alfred.command.UnmarkCommand;
+import alfred.exceptions.AlfredException;
+
+/**
+ * Represents a parser that deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Translates the user command into the respective command.
+     * @param fullCommand The full command given by the user.
+     * @return The command that is parsed out from the full command.
+     * @throws AlfredException The error thrown due to a failure to parse date and time.
+     */
+    public Command parse(String fullCommand) throws AlfredException {
+        String[] lineArr = fullCommand.split(" ");
+        String command = lineArr[0];
+        if (command.equals("bye") && lineArr.length == 1) { // So we can still add taskNames that start with bye
+            return new ExitCommand();
+        } else if (command.equals("list") && lineArr.length == 1) {
+            return new ListCommand();
+        } else if (command.equals("mark") && lineArr.length == 2) {
+            return new MarkCommand(lineArr[1]);
+        } else if (command.equals("unmark") && lineArr.length == 2) {
+            return new UnmarkCommand(lineArr[1]);
+        } else if (command.equals("delete") && lineArr.length == 2) {
+            return new DeleteCommand(lineArr[1]);
+        } else if (command.equals("find")) { // what if there's task called "find ball?"
+            return new FindCommand(lineArr[1]);
+        } else if (command.equals("list") && lineArr.length == 2) {
```

Comment:
Would using switch cases here make this code a bit easier to read?

Reply:
I agree that switch cases would make  the code a bit easier to read. Initially I was thinking of adding tasks name that include certain command names but I don't think it will be an issue anymore!